### PR TITLE
Add new deprecation cops to help people get off Chef 12

### PR DESCRIPTION
**Pull request by tas50**
Reading over the release notes we were missing these 5 breaking changes in Chef 13

Signed-off-by: Tim Smith <tsmith@chef.io>

### CODE REVIEW

**Comment by tas50** [comment]
I rant this set over a large chef repo I have and it found 656 new offenses

File: config/cookstyle.yml [hunk 1]
```diff
@@ -1062,6 +1063,46 @@ ChefDeprecations/CookbooksDependsOnSelf:
...
+  StyleGuide: '#chefdeprecationschefshellout'
+  Enabled: true
+  VersionAdded: '6.17.0'
+  Exclude:
+    - '**/metadata.rb'
+    - '**/attributes/*.rb'
+    - '**/Berksfile'
+
+ChefDeprecations/ExecutePathProperty:
+  Description: In Chef Infra Client 13 and later you must set path env vars in execute resources using the `environment` property not the legacy `path` property.
```
Comment (phiggins) [inline config/cookstyle.yml:1077]:
```suggestion
  Description: In Chef Infra Client 13 and later you must set path env vars in `execute` resources using the `environment` property not the legacy `path` property.
```

File: config/cookstyle.yml [hunk 2]
```diff
@@ -1062,6 +1063,46 @@ ChefDeprecations/CookbooksDependsOnSelf:
...
+  StyleGuide: '#chefdeprecationsexecutepathproperty'
+  Enabled: true
+  VersionAdded: '6.17.0'
+  Exclude:
+    - '**/metadata.rb'
+    - '**/attributes/*.rb'
+    - '**/Berksfile'
+
+ChefDeprecations/ExecuteRelativeCreatesWithoutCwd:
+  Description: In Chef Infra Client 13 and later you must either specific an absolute path when using the `execute` resource's `creates` property or also use the `cwd` property.
```
Comment (phiggins) [inline config/cookstyle.yml:1087]:
```suggestion
  Description: In Chef Infra Client 13 and later you must either specify an absolute path when using the `execute` resource's `creates` property or also use the `cwd` property.
```

File: spec/rubocop/cop/chef/deprecation/execute_path_property_spec.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+# limitations under the License.
+#
+
+require 'spec_helper'
+
+describe RuboCop::Cop::Chef::ChefDeprecations::ExecutePathProperty, :config do
+  subject(:cop) { described_class.new(config) }
+
+  it 'registers an offense when an execute resource has a String path property' do
+    expect_offense(<<~RUBY)\
```
Comment (phiggins) [inline spec/rubocop/cop/chef/deprecation/execute_path_property_spec.rb:24]:
```suggestion
    expect_offense(<<~RUBY)
```

?

File: spec/rubocop/cop/chef/deprecation/execute_path_property_spec.rb [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      execute 'some_cmd' do
+        path '/foo/bar'
+        ^^^^^^^^^^^^^^^ In Chef Infra Client 13 and later you must set path env vars in execute resources using the `environment` property not the legacy `path` property.
+      end
+    RUBY
+  end
+
+  it 'registers an offense when an execute resource has an Array path property' do
+    expect_offense(<<~RUBY)\
```
Comment (phiggins) [inline spec/rubocop/cop/chef/deprecation/execute_path_property_spec.rb:33]:
```suggestion
    expect_offense(<<~RUBY)
```

**Review by phiggins — APPROVED** [review]
A few small suggestions but looks good overall.

**Pull request merged**